### PR TITLE
Update cache moudle jdk17 and springboot3 

### PR DESCRIPTION
**Pull request by 3424672656**
upgrade jdk17 and springboot3.2.0 and add missing annotation

### CODE REVIEW

**Comment by 3424672656** [comment]
It's already running locally

File: 2-advanced/dubbo-samples-cache/pom.xml [hunk 1]
```diff
@@ -36,12 +36,12 @@
...
     <properties>
-        <maven.compiler.source>1.8</maven.compiler.source>
-        <maven.compiler.target>1.8</maven.compiler.target>
+        <maven.compiler.source>17</maven.compiler.source>
+        <maven.compiler.target>17</maven.compiler.target>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
 
-        <dubbo.version>3.2.6</dubbo.version>
-        <spring-boot.version>2.7.8</spring-boot.version>
+        <dubbo.version>3.3.0-beta.1</dubbo.version>
```
Comment (AlbumenJ) [inline 2-advanced/dubbo-samples-cache/pom.xml:43]:
```suggestion
        <dubbo.version>3.2.11</dubbo.version>
```

File: 2-advanced/dubbo-samples-cache/pom.xml [hunk 2]
```diff
@@ -36,12 +36,12 @@
...
+        <spring-boot.version>3.2.0</spring-boot.version>
```
Comment (AlbumenJ) [inline 2-advanced/dubbo-samples-cache/pom.xml:44]:
```suggestion
        <spring-boot.version>3.2.3</spring-boot.version>
```

**Review by AlbumenJ — CHANGES_REQUESTED** [review]

File: 2-advanced/dubbo-samples-cache/pom.xml (hunk 2 above)
Comment (3424672656) [reply 2-advanced/dubbo-samples-cache/pom.xml]: change completed

**Comment by chickenlj** [comment]
Please change replace [this dependency](https://github.com/apache/dubbo-samples/pull/1055/files#diff-337b74a6683314916ec35b3ac532137514f4c766cd7261eb2db3bfdb1bc0a529R87) with `dubbo-zookeeper-curator5-spring-boot-starter` by referencing the pom.xml file in [this sample](https://github.com/apache/dubbo-samples/blob/master/2-advanced/dubbo-samples-mock/dubbo-samples-mock-provider/pom.xml#L47)

**Comment by chickenlj** [comment]
And also `<dubbo.version>3.3.0-beta.1</dubbo.version>` is needed for this starter, so please stick to `3.3.0-beta.1` instead of `3.2.11`

File: 2-advanced/dubbo-samples-cache/pom.xml (hunk 1 above)
Comment (chickenlj) [reply 2-advanced/dubbo-samples-cache/pom.xml]: Must use `<dubbo.version>3.3.0-beta.1</dubbo.version>` in order to showcase usage of `dubbo-zookeeper-curator5-spring-boot-starter`

**Comment by 3424672656** [comment]
Please review the latest pr

**Comment by chickenlj** [comment]
> Please review the latest pr

Have you already submitted it? You can cross-reference that pr in the comment here.

**Comment by 3424672656** [comment]
Sorry for my expression error, please review the latest submission

**Comment by chickenlj** [comment]
> Sorry for my expression error, please review the latest submission

not necessary to reopen a new pr, just submitting to this PR would be fine.

**Comment by 3424672656** [comment]
Is there anything that needs to be revised now?

**Review by chickenlj — APPROVED** [review]
LGTM.

**Pull request merged**